manage_breast_screening/record_a_mammogram/views.py

```diff
@@ -1,22 +1,81 @@
 import logging
 
-from django.shortcuts import redirect, render
+from django.shortcuts import get_object_or_404, redirect, render
+from django.views.decorators.http import require_http_methods
 from django.views.generic import FormView
 
+from manage_breast_screening.clinics.models import Appointment
+
+from ..clinics.models import Appointment
 from .forms import (
     AppointmentCannotGoAheadForm,
     AskForMedicalInformationForm,
     RecordMedicalInformationForm,
     ScreeningAppointmentForm,
 )
 
+Status = Appointment.Status
+
 logger = logging.getLogger(__name__)
 
 
+def status_color(status):
+    """
+    Color to render the status tag
+    """
+    match status:
+        case Status.CHECKED_IN:
+            return ""  # no colour will get solid dark blue
+        case Status.SCREENED:
+            return "green"
+        case (Status.DID_NOT_ATTEND, Status.CANCELLED):
+            return "red"
+        case (Status.ATTENDED_NOT_SCREENED, Status.PARTIALLY_SCREENED):
+            return "orange"
+        case _:
+            return "blue"  # default blue
+
+
 class StartScreening(FormView):
     template_name = "record_a_mammogram/start_screening.jinja"
     form_class = ScreeningAppointmentForm
 
+    def get_appointment(self):
+        id = self.kwargs["id"]
+
+        return Appointment.objects.prefetch_related(
+            "clinic_slot", "screening_episode__participant"
+        ).get(pk=id)
+
+    def get_context_data(self, **kwargs):
```

Comment:
I think you already mentioned this somewhere, but this context data and the status_color function above seem like good candidates for some kind of presenter object.

Reply:
Yeah 💯 I think this will help a lot, it's on my list of things I want to chat about